content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb

```diff
@@ -1,16 +1,43 @@
+# frozen_string_literal: true
+
 #
-# Description: Get quota source.
+# Description: Get Quota Source
 #
+module ManageIQ
+  module Automate
+    module System
+      module CommonMethods
+        module QuotaMethods
+          class QuotaSource
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def main
+              quota_source
+            end
 
-@miq_request = $evm.root['miq_request']
-$evm.root['quota_source_type'] = $evm.parent['quota_source_type'] || $evm.object['quota_source_type']
+            private
 
-case $evm.root['quota_source_type'].downcase
-when 'group'
-  $evm.root['quota_source'] = @miq_request.requester.current_group
-when 'user'
-  $evm.root['quota_source'] = @miq_request.requester
-else
-  $evm.root['quota_source'] = @miq_request.tenant
-  $evm.root['quota_source_type'] = 'tenant'
+            def quota_source
+              @miq_request = @handle.root['miq_request']
```

Comment:
There is no reason to for `@miq_request` to be an instance variable.  Also, part of breaking these methods down is to separate out some of the logic into methods.

I would suggest getting rid of the `@miq_request` variable all together and creating a `miq_request` method that just gets called when needed:
```ruby
def miq_request
  @handle.root['miq_request']
end
```

Then all the callers change to use the method.  For example:
```diff
- @handle.root['quota_source'] = @miq_request.requester.current_group
+ @handle.root['quota_source'] = miq_request.requester.current_group

```

Reply:
Code refactored.